hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsScheduler.java

```diff
@@ -619,6 +626,11 @@ private void reloadConfigsOnAllRunningTasks(SchedulerDriver driver) {
   }
 
   private void correctCurrentPhase() {
+    if (!reconciler.complete()) {
```

Comment:
I don't think we need this logic here. We should start in reconciling phase and only leave once it's done.